src/main/java/it/pagopa/ecommerce/commons/domain/v1/RptId.java

```diff
@@ -20,7 +20,7 @@
  */
 @ValueObject
 public record RptId(String value) {
-    private static final Pattern rptIdRegex = Pattern.compile("([a-zA-Z\\d]{29})");
+    private static final Pattern rptIdRegex = Pattern.compile("^([0-9]{29})$");
```

Comment:
[`Matcher#matches`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Matcher.html#matches()) (used below) already matches the whole string, so `^` and `$` should not be necessary